components/mediation-admin/org.wso2.carbon.rest.api/src/main/java/org/wso2/carbon/rest/api/service/RestApiAdmin.java

```diff
@@ -1097,7 +1099,9 @@ public String getSwaggerDocument(String apiName, int tenantId) throws APIExcepti
             }
 
             try {
-                JSONObject jsonDefinition = new JSONObject(new GenericApiObjectDefinition(api).getDefinitionMap());
+                AxisConfiguration axisConfiguration = ConfigHolder.getInstance().getAxisConfiguration();
+                ServerConfig serverConfig = new CarbonServerConfig(axisConfiguration);

```

Comment:
get instance using a singleton if possible

Reply:
Since there is already common singleton holding Axis config, we don't need to make this singleton we could use that within CarbonServerConfig. Did improvement in latest commit